Nodejs/Product/Nodejs/TypingsAcquisition.cs

```diff
@@ -0,0 +1,123 @@
+            return await DownloadTypings(
+                pathToRootNpmDirectory,
+                pathToRootProjectDirectory,
+                newPackages,
+                redirector);
+        }
+
+        private static async Task<bool> DownloadTypings(
+            string pathToRootNpmDirectory,
+            string pathToRootProjectDirectory,
+            IEnumerable<IPackage> packages,
+            Redirector redirector) {
+            if (!packages.Any()) {
+                return true;
+            }
+
+            var tsdPath = GetTsdPath(pathToRootNpmDirectory);
+            if (tsdPath == null) {
+                if (redirector != null)
+                    redirector.WriteErrorLine(SR.GetString(SR.TsdNotInstalledError));
+                return false;
+            }
+
+            using (var process = ProcessOutput.Run(
+                tsdPath,
+                TsdInstallArguments(packages),
+                pathToRootProjectDirectory,
+                null,
+                false,
+                redirector,
+                quoteArgs: true)) {
+                if (!process.IsStarted) {
+                    // Process failed to start, and any exception message has
+                    // already been sent through the redirector
+                    if (redirector != null) {
+                        redirector.WriteErrorLine("could not start tsd");
+                    }
+                    return false;
+                }
+                var i = await process;
+                if (i == 0) {
+                    if (redirector != null) {
+                        redirector.WriteLine(SR.GetString(SR.TsdInstallCompleted));
+                    }
+                    return true;
+                } else {
+                    process.Kill();
+                    if (redirector != null) {
+                        redirector.WriteErrorLine(SR.GetString(SR.TsdInstallErrorOccurred));
+                    }
+                    return false;
+                }
+            }
+        }
+
+        private static string GetTsdPath(string pathToRootNpmDirectory) {
+            var path = Path.Combine(pathToRootNpmDirectory, TsdExe);
+            return File.Exists(path) ? path : null;
+        }
+
+        private static string GetPackageTsdName(IPackage package) {
+            return package.Name;
+        }
+
+        private static IEnumerable<string> TsdInstallArguments(IEnumerable<IPackage> packages) {
+            return new[] { "install", }.Concat(packages.Select(GetPackageTsdName)).Concat(new[] { "--save" });
+        }
+
+        private static IEnumerable<IPackage> TypingsToAcquire(string pathToRootProjectDirectory, IEnumerable<IPackage> packages) {
+            var currentTypings = new HashSet<string>(CurrentTypingsPackages(pathToRootProjectDirectory));
+            return packages.Where(package =>
+                !currentTypings.Contains(GetPackageTsdName(package)));
```

Comment:
This consistently defines how we map IPackages to strings. It doesn't matter that the implementation is a single line.

This sort of abstraction is useful because when looking at the code, since you only have to know that we map the IPackage to a string somehow. This also makes any future logic changes (such as including the version as part of the string) much easier and safer since we only have to update a single function, instead of every consumer of this logic.